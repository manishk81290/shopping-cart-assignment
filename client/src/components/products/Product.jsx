import React, { useState, useEffect } from "react";
import { Row, Col, Alert } from "react-bootstrap";
import ProductItem from "./ProductItem";
import ShoppingContext from "../../context/ShoppingContext";

import "./product.scss";

const Product = ({ match }) => {
  const shoppingContext = React.useContext(ShoppingContext);
  const { products, getProducts } = shoppingContext;
  const [product, setProduct] = useState(products);
  useEffect(() => {
    if (!products.length) getProducts();
  }, []);
  useEffect(() => {
    if (match.params.productCategory === "all") {
      setProduct((prevProduct) => products);
    } else {
      setProduct((prevProduct) =>
        products.filter(
          (prod) => prod.category === match.params.productCategory
        )
      );
    }
    console.log(product);
  }, [match.params.productCategory, products]);
  return (
    <Col lg={9} md={9} sm={12}>
      <Row lg={9} md={12} sm={12}>
        {product.length ? (
          product.map((prod) => {
            return <ProductItem item={prod} key={prod.key} />;
          })
        ) : (
          <Alert variant="info">No Product Found!!!!</Alert>
        )}
      </Row>
    </Col>
  );
};

export default Product;
