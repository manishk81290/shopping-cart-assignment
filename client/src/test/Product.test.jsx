import React from "react";
import Product from "./../components/products/Product";
import ShallowRenderer from "react-test-renderer/shallow";

let realUseContext;
let useContextMock;
const sampleContextData = {
  products: [
    {
      name: "Capsicum - Green, 1 kg",
      imageURL: "/static/images/products/fruit-n-veg/capsicum-green.jpg",
      description:
        "Leaving a moderately pungent taste on the tongue, Green capsicums, also known as green peppers are bell shaped, medium-sized fruit pods.",
      price: 137,
      stock: 50,
      category: "5b6899953d1a866534f516e2",
      sku: "fnw-capsicum-1",
      id: "5b6c6bdc01a7c38429530886",
      count: 2,
    },
  ],
  getProducts: () => {},
};

const testProps = {
  match: {
    params: {
      productCategory: "5b6c6f4a01a7c3842953088c",
    },
  },
};

beforeEach(() => {
  realUseContext = React.useContext;
  useContextMock = React.useContext = jest.fn();
});
// Cleanup mock
afterEach(() => {
  React.useContext = realUseContext;
});

it("Product with mock useContext hook", () => {
  useContextMock.mockReturnValue(sampleContextData);
  const element = new ShallowRenderer().render(<Product {...testProps} />);
  expect(element).toBeTruthy();
  expect(element).toMatchSnapshot();
});
