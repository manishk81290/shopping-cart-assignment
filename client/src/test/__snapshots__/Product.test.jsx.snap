// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product with mock useContext hook 1`] = `
<Col
  lg={9}
  md={9}
  sm={12}
>
  <Row
    lg={9}
    md={12}
    noGutters={false}
    sm={12}
  >
    <ProductItem
      item={
        Object {
          "category": "5b6899953d1a866534f516e2",
          "count": 2,
          "description": "Leaving a moderately pungent taste on the tongue, Green capsicums, also known as green peppers are bell shaped, medium-sized fruit pods.",
          "id": "5b6c6bdc01a7c38429530886",
          "imageURL": "/static/images/products/fruit-n-veg/capsicum-green.jpg",
          "name": "Capsicum - Green, 1 kg",
          "price": 137,
          "sku": "fnw-capsicum-1",
          "stock": 50,
        }
      }
    />
  </Row>
</Col>
`;
